### PR TITLE
Chores

### PR DESCRIPTION
**Pull request by lyca**
Update versions while still keeping it compatible with Java 8 (Spring Boot >= 3.x needs Java 17).

### CODE REVIEW

**Review by YunaBraska — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -101,7 +101,7 @@ To depend on the autoconfigure module, simply add it as a dependency in your pom
 <dependency>
     <groupId>io.nats</groupId>
     <artifactId>jnats</artifactId>
-    <version>2.12.0</version>
+    <version>2.16.13</version>
```
Comment (scottf) [inline README.md:104]: What about 2.17.1?

File: nats-spring-boot-starter/pom.xml [hunk 2]
```diff
@@ -28,7 +28,7 @@
         <dependency>
             <groupId>io.nats</groupId>
             <artifactId>jnats</artifactId>
-            <version>2.12.0</version>
+            <version>2.16.13</version>
```
Comment (scottf) [inline nats-spring-boot-starter/pom.xml:31]: What about 2.17.1?

File: nats-spring-cloud-stream-binder/pom.xml [hunk 3]
```diff
@@ -27,7 +27,7 @@
         <dependency>
             <groupId>io.nats</groupId>
             <artifactId>jnats</artifactId>
-            <version>2.12.0</version>
+            <version>2.16.13</version>
```
Comment (scottf) [inline nats-spring-cloud-stream-binder/pom.xml:30]: What about 2.17.1?

File: nats-spring/pom.xml [hunk 4]
```diff
@@ -26,7 +26,7 @@
         <dependency>
             <groupId>io.nats</groupId>
             <artifactId>jnats</artifactId>
-            <version>2.12.0</version>
+            <version>2.16.13</version>
```
Comment (scottf) [inline nats-spring/pom.xml:29]: What about 2.17.1?

**Review by scottf — CHANGES_REQUESTED** [review]
2.17.1 instead of 2.16.13

**Comment by scottf** [comment]
Is there something specific about 2.16.13 that it should be used instead of 2.17.1?

**Pull request merged**